inngest-core/src/main/kotlin/com/inngest/Step.kt

```diff
@@ -77,4 +84,43 @@ class Step(val state: State) {
             throw StepInterruptSleepException(id, hashedId, "${durationInSeconds}s")
         }
     }
+
+    /**
+     * Sends multiple event to Inngest
+     *
+     * @param id Unique step id for memoization.
+     * @param event An event payload object.
+     *
+     */
+
+    fun sendEvent(
+        id: String,
+        event: InngestEvent,
+    ) = sendEvent(id, arrayOf(event))
+
+    /**
+     * Sends a single event to Inngest
+     *
+     * @param id Unique step id for memoization.
+     * @param events An array of event payload objects.
+     *
+     */
+    fun sendEvent(
+        id: String,
+        events: Array<InngestEvent>,
+    ) {
+        val hashedId = state.getHashFromId(id)
+
+        try {
+            // If this doesn't throw an error, it's null and that's what is expected
+            val stepState = state.getState<Any?>(hashedId)
+            if (stepState != null) {
+                throw Exception("step state expected sleep, got something else")
+            }
+            return
+        } catch (e: StateNotFound) {
+            val response = CommHandler.sendEvent<SendEventsResponse>(events)
+            throw StepInterruptSendEventException(id, hashedId, response!!.ids)
```

Comment:
minor, let's see if we can change the return type of `CommHandler.sendEvent` to guarantee non null response

Reply:
we'll investigate this later too